Sources/Get/Request.swift

```diff
@@ -11,22 +11,22 @@ import FoundationNetworking
 /// An HTTP network request.
 public struct Request<Response>: @unchecked Sendable {
     /// HTTP method, e.g. "GET".
-    public var method: String
+    public var method: HTTPMethod
     /// Resource URL. Can be either absolute or relative.
-    public var url: String
+    public var url: URL?
     /// Request query items.
     public var query: [(String, String?)]?
```

Comment:
What if this was `[URLQueryItem]?` on `Request` but with helper inits? Mainly cause I personally don't work much with tuples.

Reply:
I also considered it, and I'm not sure yet. Tuples are probably not the ideal choice for this case. When I think about tuples, I usually think about a bag of values that are loosely related and there is no good name for them. That's not the case for the query items.

```swift
var request = Request(path: "/domain.tld")
request.query = [("name", "value1+value2")]
request.query = [URLQueryItem(name: "name", value: "value1+value2")]
```